pennylane/qchem/integrals.py

```diff
@@ -304,13 +319,22 @@ def _overlap_integral(*args):
         Returns:
             array[float]: the overlap integral between two contracted Gaussian orbitals
         """
-
         args_a = [arg[0] for arg in args]
         args_b = [arg[1] for arg in args]
+
+        # Generate params returns params in alpha, coeff, coord order
         alpha, ca, ra = _generate_params(basis_a.params, args_a)
         beta, cb, rb = _generate_params(basis_b.params, args_b)
 
-        if basis_a.params[1].requires_grad or normalize:
+        if (
+            getattr(basis_a.params[1], "requires_grad", False)
+            or normalize
+            or (
+                len(args) > 0
+                and qml.math.get_interface(basis_a.params[1]) == "jax"
+                and qml.math.requires_grad(args[1])
+            )
+        ):
```

Comment:
Mhm I thought about doing that. My draft solution: since this check here is not consistent across all the integrals, e.g. some of them want to check args[1] some want to check args[0] and etc... the solution would be a function `check_requires_grad(index)` where index is the arg you check. No issues with that right?